### ISSUE TITLE
Qusetion about the processed data from ShapeNet

### ISSUE THREAD

**Issue by hmax233** [czq142857/BSP-NET-pytorch#12]
Hi ZhiQin!

Thanks for your work and releasing scripts.
I want to know the way you prepare the voxel data, I notice that  there are "data_voxels" and "coords" needed for the 3d AE.
The "data_voxels" are the model voxel martrixs and the "coords" are float coordinates of the voxels. The "coords" are splited to 8 subsets, values of which are limited to[-0.4922, 0.4766]. After you get the point clouds from ShapeNet, how you normalize the data and how you voxelize the data after normalization?

Best,
hmax

**Comment by czq142857** [comment]
Please use the provided data preparation code.

If you want to create voxels and convert them into training data from scratch, see these threads:
https://github.com/czq142857/IM-NET/issues/10
https://github.com/czq142857/IM-NET/issues/19

**Comment by hmax233** [comment]
Thanks a lot for your favour.
I find these threads are mostly about converting mesh model to binvoxel. I want to know how to voxelize the point clouds to the binvoxel which the bspnet needs.

**Comment by czq142857** [comment]
The training data of BSP-net is not obtained from point clouds. It is from voxels. So you should prepare either meshes or voxels as your "raw" data, and use the provided data preparation code to convert them into the training data.

**Comment by hmax233** [comment]
![图片](https://user-images.githubusercontent.com/58834147/183848625-ac4e1f12-c1d1-4be7-90c3-702f72763115.png)
![图片](https://user-images.githubusercontent.com/58834147/183848716-ac1e30d1-70c5-4e04-812e-507deadbc422.png)
I'm appreciated for your suggestions,which inspire me!
I only use the point cloud from the mesh model and put the point cloud into a bbox( min=[-0.5, -0.5, -0.5], max = [0.5, 0.5, 0.5] ). Points over the bounding box are discarded. Then I voxelize the point clouds to get the binvoxel matrix(size = [1,1,64,64,64], values = 1 means not empty). And I inference the BSPNet with the binvoxel matrix,which export the the  second image.The first image is the input mesh model.Now I know it's works!
But the reconstruction quality seems to be unsatisfying.Do you have some  suggestions for that?

**Comment by czq142857** [comment]
First, you should not use the voxels obtained from point clouds. The training voxels are solid, but your voxels are probably hollow, as the point clouds only contain points on the surface, not inside.

Second, if you are using the pretrained weights, you should normalize the shape so that the diagonal of the bounding box is 1, i.e., H^2+W^2+D^2 = 1, not max(H,W,D) = 1.

**Comment by hmax233** [comment]
Thanks for your suggestions,I change the way I normalize the shape, Now I divide the point cloud by the diagonal length, and then voxelize the space of size [-0.5,-0.5,-0.5]-> [0.5,0.5,0.5]. The reconstruction quality improves. Am I now voxelizing the right way? About the problem that point clouds are hollow,I consider to train the BSPNet in the dataset obtained from point clouds.

**Comment by czq142857** [comment]
If your data is from point clouds, you might need to modify the data preparation code to accommodate that. BSP-net is a kind of neural implicit and neural implicit requires the shapes to be solid.

**Comment by czq142857** [comment]
Point cloud inputs are fine. But to train the model from scratch, you still need watertight, solid shapes.
